waku/v2/protocol/waku_filter/waku_filter_types.nim

```diff
@@ -15,21 +15,24 @@ const
   MaxRpcSize* = 10 * MaxWakuMessageSize + 64*1024
 
 type
+  Topic* = string

```

Comment:
This type causes a name colision (see my comment above).
Maybe call it PubSubTopic to make clear it is not the ContentTopic.

Wakunode2.nim and filter could use the same type. You could  use the type declared here in filter also in wakunode2.

Also, if you want to make sure it is not any kind of string but this type in specific, you coud use

```nim
type
  PubSubTopic* = distinct string
```

Reply:
I agree that we would need to have:
* either a shared `PubSubTopic` and `ContentTopic` type aliases between all modules
* or a "module-level" type aliases and one at the waku node module level that are compatible but not interfering

Reply:
WDYT of adding this type (`PubSubTopic`) export to the `waku_message` module? Although not strictly part of the message, it is so related in concept that I'd be comfortable having it there. This is also where we export `ContentTopic*` which would make the difference and similarity a bit more apparent and it already serves as the "common types" module between all protocols. We could even consider renaming the module to common_types or something, but I don't feel that's necessary.

Reply:
We need to give it a thought. Maybe I can tackle it as part of my next PR, and live it for now as @kaiserd suggested.

Reply:
Sounds reasonable to me.

Reply:
I agree on adding PubSubTopic to `waku_message`  in a follow up PR. My first intuition was searching the type in waku_message :)